bindings/python/htcondor2/_negotiator.py

```diff
@@ -1 +1,180 @@
+from typing import List
 
+from ._common_imports import (
+    classad,
+    Collector,
+    DaemonType,
+)
+
+from .htcondor2_impl import (
+    _negotiator_command,
+    _negotiator_command_return,
+    _negotiator_command_user,
+    _negotiator_command_user_return,
+    _negotiator_command_user_value,
+)
+
+
+class Negotiator():
+
+    def __init__(self, location : classad.ClassAd = None):
+        if location is None:
+            c = Collector()
+            location = c.locate(DaemonType.Negotiator)
+
+        if not isinstance(location, classad.ClassAd):
+            raise TypeError("location must be a ClassAd")
+
+        self._addr = location['MyAddress']
+        # We never actually use this for anything.
+        # self._version = location['CondorVersion']
+
+
+    def deleteUser(self, user : str) -> None:
+        if not isinstance(user, str):
+            raise TypeError("user must be a string")
+        if '@' not in user:
+            # This was HTCondorValueError in version 1.
+            raise ValueError("You must specify the submitter (user@uid.domain)")
+        _negotiator_command_user(self._addr, self._DELETE_USER, user)
+
+
+    def getPriorities(self, rollup : bool) -> List[classad.ClassAd]:
+        command = self._GET_PRIORITY
+        if rollup:
+            command = self._GET_PRIORITY_ROLLUP
+        returnAd = _negotiator_command_return(self._addr, command)
+        return _convert_numbered_attributes_to_list_of_ads(returnAd)
+
+
+    def getResourceUsage(self, user : str) -> List[classad.ClassAd]:
+        if not isinstance(user, str):
+            raise TypeError("user must be a string")
+        if '@' not in user:
+            # This was HTCondorValueError in version 1.
+            raise ValueError("You must specify the submitter (user@uid.domain)")
+        returnAd = _negotiator_command_user_return(self._addr, self._GET_RESLIST, user)
+        return _convert_numbered_attributes_to_list_of_ads(returnAd)
+
+
+    def resetAllUsage(self):
+        _negotiator_command(self._addr, self._RESET_ALL_USAGE)
+
+
+    def resetUsage(self, user : str) -> None:
+        if not isinstance(user, str):
+            raise TypeError("user must be a string")
+        if '@' not in user:
+            # This was HTCondorValueError in version 1.
+            raise ValueError("You must specify the submitter (user@uid.domain)")
+        _negotiator_command_user(self._addr, self._RESET_USAGE, user)
+
+
+    def setBeginUsage(self, user : str, when : int) -> None:
+        if not isinstance(user, str):
+            raise TypeError("user must be a string")
+        if '@' not in user:
+            # This was HTCondorValueError in version 1.
+            raise ValueError("You must specify the submitter (user@uid.domain)")
+        _negotiator_command_user_value(self._addr, self._SET_BEGINTIME, user, int(when))
```

Comment:
Wondering if we should throw a warning if `when` and/or `ceiling` are floats, though I would expect those fractions to be insignificant in the grand scheme.

Reply:
We should probably do that at some point, but for now I'd rather not break existing code.